source/Core/Threads/GUIThread.cpp

```diff
@@ -860,13 +860,12 @@ void startGUITask(void const *argument) {
   getTipRawTemp(1); // reset filter
   OLED::setRotation(getSettingValue(SettingsOptions::OrientationMode) & 1);
   uint32_t ticks = xTaskGetTickCount();
-  ticks += (TICKS_SECOND * 4); // 4 seconds from now
-  while (xTaskGetTickCount() < ticks) {
+  ticks += (getSettingValue(SettingsOptions::LOGOTime) / 100 * TICKS_SECOND);
```

Comment:
You would probably want to do the divide and multiply in the other order.
As this is an integer, you want to go bigger then smaller to preserve accuracy.